client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js

```diff
@@ -11,81 +11,130 @@ import {
   declarative_charts,
 } from "../../shared.js";
 import { Canada } from '../../../../charts/canada/index.js';
+import { SlideToggle, SpinnerWrapper } from '../../../../components/index.js';
+import { get_static_url, make_request } from '../../../../request_utils.js';
 
 const { std_years } = year_templates;
+
 const formatter = formats["compact2_raw"];
 
 const { text_maker, TM } = create_text_maker_component(text);
 const { provinces, provinces_with_article } = businessConstants;
 const { A11YTable } = declarative_charts;
 
-const calculate_common = (subject, table) => {
-  const level = subject.level;
 
-  if ( level === 'dept' && !_.has(table.depts , subject.id) ){
-    return false;
+function loadPopulation(){
+  const parse_csv_string = csv_string => _.tail( d3.csvParseRows( _.trim(csv_string) ) );
+
+  const load_csv = () => make_request( get_static_url(`csv/population.csv`) )
+    .then( csv_string => parse_csv_string(csv_string) );
+    
+  return load_csv().then(function(loaded) {
+    const transformed = _.chain(loaded)
+      .keyBy(row => row[0])
+      .mapValues(row => _.chain(row)
+        .tail()
+        .map(value => parseInt(value))
+        .value()
+      )
+      .value();
+    
+    return transformed;
   }
+  ); 
```

Comment:
Putting aside the two comments above, and focusing on the lodash chain here, the same result could be obtained with
```
const transformed = _.chain(loaded)
  .map( ([prov_code, ...values]) => [
    prov_code, 
    [_.map(values, parseInt)],
  ])
  .fromPairs()
  .value()
```

Not that one way's surpassingly better. I just tend to use `fromPairs` instead of `keyBy` in pretty much any case (`keyBy` often leaves you with the headache of following up by dropping whatever was used as the key). Mostly just bringing this up because you'll be reading a lot of my code and this was an excuse to point out why I write a lot of `fromPairs`.

Reply:
Changed to your fromPairs method, it seemed to make more linear sense
Done